src/txrebroadcast.cpp

```diff
@@ -0,0 +1,219 @@
+// Copyright (c) 2021 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <chainparams.h>
+#include <consensus/consensus.h>
+#include <logging.h>
+#include <miner.h>
+#include <script/script.h>
+#include <txrebroadcast.h>
+#include <util/time.h>
+#include <validation.h>
+
+#include <boost/multi_index/hashed_index.hpp>
+#include <boost/multi_index/member.hpp>
+#include <boost/multi_index/ordered_index.hpp>
+#include <boost/multi_index_container.hpp>
+
+/** We rebroadcast 3/4 of max block weight to reduce noise due to circumstances
+ *  such as miners mining priority transactions. */
+static constexpr unsigned int MAX_REBROADCAST_WEIGHT{3 * MAX_BLOCK_WEIGHT / 4};
+
+/** Default minimum age for a transaction to be rebroadcast */
+static constexpr std::chrono::minutes REBROADCAST_MIN_TX_AGE{30min};
+
+/** Maximum number of times we will rebroadcast a transaction */
+static constexpr int MAX_REBROADCAST_COUNT{6};
```

Comment:
> At that point, should we return an event to the wallet logic inviting to RBF the transaction ?

that's a cool idea. I don't think it is applicable yet, but I've noted it down for future work where it would be more relevant.

why I don't think it is applicable yet: as of this patch, `ResendWalletTransactions` calls `SubmitMemoryPoolAndRelay` with the `relay` bool set to `true`. This gets used in `BroadcastTransaction` to force relaying the transaction to peers at the time of submission. in the future, the goal is to disable this (see "Merge Plan & Next Steps" in  https://github.com/bitcoin/bitcoin/pull/21061#issuecomment-771426879), at which time I believe this sort of mechanism would make sense.